.github/workflows/build-github-cli-image.yaml

```diff
@@ -0,0 +1,44 @@
+name: Build and publish github-cli image to ECR
+
+on:
+  workflow_dispatch:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+    paths:
+      - "docker/images/github-cli/Dockerfile"
+
+jobs:
+  publish:
```

Comment:
Is there a reason you can't just call https://github.com/alphagov/govuk-infrastructure/blob/main/.github/workflows/ci-ecr.yaml here?

Reply:
Good call - I've updated the ci-ecr workflow to accept inputs to make it also usable in this context (e.g. ecr repo name was hardcode to the git repo name etc). I've also applied your suggestions above to the ci-ecr workflow.

Reply:
Awesome, thanks!